### PR TITLE
QueryOptions: update maxDataPoints text and show any value that is configured

### PR DESCRIPTION
**Pull request by ryantxu**
The query UI will show an options editor for features that are defined in the `plugin.json` -- this PR:
* shows options that are configured (saved in the panel) even if the datasource does not say it cares.  `maxDataPoints` is actually used on most datasources, although few allow setting the option
* updates the help text for `maxDataPoints` to explain how it is used in streaming
* removes `buffer` from testdata stream query and uses `maxDataPoints`

### CODE REVIEW

File: public/app/features/dashboard/panel_editor/QueryOptions.tsx [hunk 1]
```diff
@@ -70,8 +70,9 @@ export class QueryOptions extends PureComponent<Props, State> {
       name: 'maxDataPoints',
       tooltipInfo: (
         <>
-          The maximum data points the query should return. For graphs this is automatically set to one data point per
-          pixel.
+          The maximum data points the query should return. This is used to calculate a reasonable interval that is
```
Comment (torkelo) [inline public/app/features/dashboard/panel_editor/QueryOptions.tsx]: it's actually not true, I would love to figure out an away to support maxDataPoints for every data source (those that use interval), think maxDataPoints should override the interval
https://github.com/grafana/grafana/blob/master/public/app/features/dashboard/state/PanelQueryRunner.ts#L172

**Comment by ryantxu** [comment]
@torkelo -- I changed the interval calculation so it is based on maxDataPoints (that makes sense to me).  Alternatively, I could change the text and leave the behavior based on width.  But that is problematic for other reasons: See #16479

With this change, I think it might make sense to *always* show maxDataPoints, but make it clear it is a "hint" for the DataSource

**Comment by torkelo** [comment]
This needs a little more thought now. 

I would like to enable maxDataPoints for all datasources that use "minInterval". the biggest problem is that it will be a breaking change. 

As the singlestat (old) panel set maxDataPoints to 100. So this option will be set and would change the interval for a bunch of influxdb & prometheus dashboards that use singlestat. 

Not sure what the solution for that is or how much of a problem is.

**Comment by ryantxu** [comment]
@torkelo -- I changed this so that it no longer affects the `minInterval` calculation.  This now just:
1. adds a note about how `maxDataPoints` gets used for streaming buffer size
2. will show any of the settings if they exist -- regardless of the ds config
3. adds `maxDataPoints` to the testdata confg

**Review by torkelo — APPROVED** [review]

**Pull request merged**